rclcpp_components/doc/architecture.rst

```diff
@@ -0,0 +1,263 @@
+
+The C++ base type name will always be ``rclcpp_components::NodeFactory``.
+
+Registering with ``pluginlib``
+------------------------------
+
+To register with ``pluginlib`` you need to provide the "plugin category" and the plugin description file.
+The plugin category is just the name which will be used when looking for plugins of a certain at runtime.
+By convention and specifically for composable nodes based on ``rclcpp``, it should be ``rclcpp_components``.
+
+This is done with a CMake macro from ``pluginlib`` itself called ``pluginlib_export_plugin_description_file()``, see:
+
+https://github.com/ros/pluginlib/blob/c0bbfaf8b22f3800bc3bd20414b9b15ea0aa52de/pluginlib/cmake/pluginlib_export_plugin_description_file.cmake#L22-L81
+
+The ``rclcpp_components_register_node_component_library()`` will do this step for you.
+
+Creating a Stand-alone Node Executable
+--------------------------------------
+
+This package also provides some "boilerplate" source code that can be compiled into a main executable for each of your nodes, if you desire that.
+This generic main function is templated to take the salient information on how to locate your node (name from the plugin description, and your package which provides that composable node), and it uses that information at runtime to either load and instantiate your node by itself in the executable or contact an already running composable node container process and have that process load and execute your node.
+
+Composable Node Container
+-------------------------
+
+This package provides one or more "composable node containers", which exist to assist with loading and executing multiple composable nodes in a single process without the need to write your own main function to do this.
+
+The containers could receive the information about which nodes to load and execute in various ways, but only two are discussed here: a configuration file or via a ROS interface.
+
+In both cases, the information related to loading and executing a node consists the following information (some required and some optional):
+
+- package name
+  - name of the package in which the node is located
+- node plugin name
+  - the name for the plugin that was given in the ``rclcpp_components_describe_node_component()`` or the plugin description XML
+- node name
+  - overrides the node name hard coded into source code (i.e. like ``__node:=<new name>`` on the command line)
+- node namespace
+  - overrides the node namespace hard coded into the source code (i.e. like ``__ns:=<new ns>`` on the command line)
+- remappings
+  - remappings that apply to this node for things like topics and services
+- initial parameter values
+  - initial values for required parameters or to be used instead of the defaults for optional parameters
+
+Essentially, where to find the node (package name plus node name) and anything that can be influenced with command line arguments to nodes normally.
+
+Compose Nodes with a Configuration File
```

Comment:
Is this currently implemented in the container?  It doesn't seem like it to me, but I could be wrong.